### PR TITLE
ListTextHandler allowed_values_function support fixed

### PR DESCRIPTION
**Pull request by kugta**
When a list field uses `allowed_values_function` instead of `allowed_values` the value cannot be set with the current logic neither by passing the key or the value of the allowed value(s). This pull request addresses this issue.

Also refactored so it behaves the same way for both `allowed_values_function` and `allowed_values` use cases. Supports both matches in keys or values of the allowed values. This implementation explicitly searches for a value if it is allowed, unlike the previous solution which was defaulted to the `value` in case there was no match for `key`.

### CODE REVIEW

**Comment by janoka** [comment]
@kugta ++

**Comment by jhedstrom** [comment]
Sorry for the delay in reviewing this. Any chance you can rebase and resolve the conflicts?

**Comment by kugta** [comment]
Thanks @jhedstrom I suspect the issue was already resolved, I will make a few tests to confirm and close my PR.

**Comment by joaocsilva** [comment]
I still have the same issue, can you @kugta re-open the PR please?

**Comment by kugta** [comment]
@joaocsilva I have re-opened and resolved the conflicts, could you please test and confirm that it works as described above? Thanks!

File: src/Drupal/Driver/Fields/Drupal7/ListTextHandler.php [hunk 1]
```diff
@@ -16,19 +16,19 @@ public function expand($values) {
...
       $callback = $this->fieldInfo['settings']['allowed_values_function'];
-      $fn_allowed_values = call_user_func($callback, $this->fieldInfo, $this, $this->entityType, $this->entity, $cacheable);
-      $options = array_flip($fn_allowed_values);
+      $options = call_user_func($callback, $this->fieldInfo, $this, $this->entityType, $this->entity, $cacheable);
     }
     else {
-      $options = array_flip($this->fieldInfo['settings']['allowed_values']);
+      $options = $this->fieldInfo['settings']['allowed_values'];
     }
+    $allowed_values = array();
```
Comment (joaocsilva) [inline src/Drupal/Driver/Fields/Drupal7/ListTextHandler.php]: The `$allowed_values` is already declared in line 15.

**Comment by joaocsilva** [comment]
@kugta , for me works flawless :+1:  thanks

**Comment by kugta** [comment]
Thanks @joaocsilva, I removed the redundancy!
@jhedstrom would you please consider this now?

**Comment by jhedstrom** [comment]
Awesome. Thanks!

**Comment by joaocsilva** [comment]
Hi @jhedstrom , do you have planned a new release? 
We would like to have the new code, if the release will happen soon we can wait otherwise we have to reference the commit.
Thank you.

**Comment by jhedstrom** [comment]
@joaocsilva I just released alpha3

**Comment by joaocsilva** [comment]
@jhedstrom  perfect, thank you :+1:

**Pull request merged**